src/ape/api/providers.py

```diff
@@ -268,69 +360,184 @@ class ProviderAPI:
+        """
 
     @abstractmethod
     def get_nonce(self, address: str) -> int:
-        ...
+        """
+        Get the number of times an account has transacted.
+
+        Args:
+            address (str): The address of the account.
+
+        Returns:
+            int
+        """
 
     @abstractmethod
     def estimate_gas_cost(self, txn: TransactionAPI) -> int:
-        ...
+        """
+        Estimate the cost of gas for a transaction.
+
+        Args:
+            txn (:class:`~ape.api.providers.TransactionAPI`):
+                The transaction to estimate the gas for.
+
+        Returns:
+            int: The estimated cost of gas.
+        """
 
     @property
     @abstractmethod
     def gas_price(self) -> int:
-        ...
+        """
+        The price for what it costs to transact.
+
+        Returns:
+            int
+        """
 
     @property
     def priority_fee(self) -> int:
+        """
+         A miner tip to incentivize them
+        to include your transaction in a block.
+        Only providers that implement EIP-1559 will have this property,
+        otherwise will raise `NotImplementedError`.
+
+        Returns:
+            int: Value of fee.
+        """
         raise NotImplementedError("priority_fee is not implemented by this provider")
 
     @property
     def base_fee(self) -> int:
+        """
+        The minimum value required to get your transaction
+        included on the next block.
+        Only providers that implement EIP-1559 will use this property,
+        otherwise will raise ``NotImplementedError``.
+
+        Returns:
+            int
+        """
         raise NotImplementedError("base_fee is not implemented by this provider")
 
     @abstractmethod
     def get_block(self, block_id: BlockID) -> BlockAPI:
-        ...
+        """
+        Get a block.
+
+        Args:
+            block_id (:class:`~ape.types.blockID`): The ID of the block to get.
+                Can be `latest`, `earliest`, `pending`, a block hash or a block number.
+
+        Returns:
+            :class:`~ape.types.BlockID`: The block for the given ID.
+        """
 
     @abstractmethod
     def send_call(self, txn: TransactionAPI) -> bytes:  # Return value of function
-        ...
+        """
+        Execute a new transaction call immediately without creating a
+        transaction on the block chain.
+
+        Args:
+            txn: :class:`~ape.api.providers.TransactionAPI`
+
+        Returns:
+            str: The result of the transaction call.
+        """
 
     @abstractmethod
     def get_transaction(self, txn_hash: str) -> ReceiptAPI:
-        ...
+        """
+        Get the information about a transaction requested by transaction hash.
+
+        Args:
+            txn_hash (str): The hash of the transaction to retrieve.
+
+        Returns:
+            :class:`~api.providers.ReceiptAPI`:
+                The receipt of the transaction with the given hash.
```

Comment:
Does this indentation work when you build it?

I thought we figured out that we cannot indent returns statement docs here.

Reply:
Indent does not work here, you are right